### PR TITLE
Updating source for ES 5.2.1 and adding tests

### PR DESCRIPTION
**Pull request by apanimesh061**
Hi,
Following are the list of changes:
- Updated source to make it compatible with `ES 5.2.1`
- Refactoring in packages
- Added unit tests
- Updated `es-plugin.properties` with more information
- Updated versions of plugins in `pom.xml`

Thanks.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,95 +1,132 @@
...
-	</build>
-	<dependencies>
-		<dependency>
-			<groupId>org.elasticsearch</groupId>
-			<artifactId>elasticsearch</artifactId>
-			<version>${elasticsearch.version}</version>
-			<scope>provided</scope>
-		</dependency>
-	</dependencies>
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
Comment (apanimesh061) [inline pom.xml]: Indentation has been changed

File: pom.xml [hunk 2]
```diff
@@ -1,95 +1,132 @@
...
+                        <goals>
+                            <goal>single</goal>
+                        </goals>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <dependencies>
```
Comment (apanimesh061) [inline pom.xml]: Added more dependencies

File: src/main/java/elasticsearch/concatenate/ConcatenatePlugin.java [hunk 3]
```diff
@@ -1,20 +0,0 @@
-package elasticsearch.concatenate;
```
Comment (apanimesh061) [inline src/main/java/elasticsearch/concatenate/ConcatenatePlugin.java]: File moved from `elasticsearch.concatenate` to `org.elasticsearch.plugin.analysis.concatenate` package.

**Comment by francesconero** [comment]
Hello @apanimesh061! Sorry for the delay, I am not mantaining this plugin at this time, would you like to become a collaborator to the project?

**Pull request closed without merging**